lib/inaturalist_api_v2.js

```diff
@@ -697,31 +697,31 @@ const InaturalistAPIV2 = class InaturalistAPIV2 {
       paths: "./openapi/paths/v2",
       promiseMode: true,
       securityHandlers: {
-        userJwtRequired: req => InaturalistAPIV2.userJwtValidate( req, "required" ).then( ( ) => {
+        userJwtOptional: async req => {
+          await InaturalistAPIV2.verifyHeaderJWTs( req );
+          return true;
+        },
+        userJwtRequired: async req => {
+          await InaturalistAPIV2.verifyHeaderJWTs( req );
           if ( req.userSession ) {
-            return Promise.resolve( true );
+            return true;
           }
-          const jwtMissingUserError = new Error( "JWT does not specify a user" );
-          jwtMissingUserError.status = 401;
-          throw jwtMissingUserError;
-        } ).catch( err => {
-          throw err;
-        } ),
-        userJwtOptional: req => InaturalistAPIV2.userJwtValidate( req ).catch( err => {
-          throw err;
-        } ),
-        appOrUserJwtRequired: req => InaturalistAPIV2.userJwtValidate( req )
-          .then( ( ) => InaturalistAPIV2.applicationJwtValidate( req ) )
-          .then( ( ) => {
-            if ( req.userSession || req.applicationSession ) {
-              return Promise.resolve( true );
-            }
-            const jwtMissingContextError = new Error( "JWT does not specify an application or a user" );
-            jwtMissingContextError.status = 401;
-            throw jwtMissingContextError;
-          } ).catch( err => {
-            throw err;
-          } )
+          throw JWT_MISSING_OR_INVALID_ERROR;
```

Comment:
Isn't it useful to allow the client to distinguish between a missing JWT and an invalid one? Like if the JWT has expired, the client should try to request a new one, but if it's missing entirely, that might indicate some other kind of internal bug that needs to be addressed.

Reply:
That was the behavior before this PR. I wasn't sure what to do in the case of multiple tokens - maybe just return whatever the first error was. I also thought it would be easy enough for the client to know if they sent a JWT or not and that this would simplify the securityHandler code. But we can go back to two different errors if we think that distinction is helpful.

Reply:
Sounds good.